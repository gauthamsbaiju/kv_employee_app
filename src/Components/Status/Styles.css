.active{
    width: 111px;
    height: 30px;
    background: #D3F4BE ;
    border-radius: 20px;
    text-align: center;
    padding-top: 5px;
}

.inactive{  
    width: 111px;
    height: 30px;
    background: #FFBFBF 0% 0% no-repeat padding-box;
    border-radius: 20px;
    text-align: center;
    padding-top: 5px;
}