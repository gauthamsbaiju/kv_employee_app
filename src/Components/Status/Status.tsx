import React from 'react';
import './Styles.css';

export type StatusPropsType = {
  value: boolean;
};

const Status: React.FC<StatusPropsType> = (props) => {
  let cname: string = '';

  if (props.value) cname = 'active';
  // console.log(cname);
  else cname = 'inactive';
  // console.log(cname);

  return <div className={cname} data-testid='status-test'>{cname}</div>;
};

export default Status;
