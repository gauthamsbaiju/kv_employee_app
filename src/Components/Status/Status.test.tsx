import Status, { StatusPropsType } from './Status';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Test for status', () => {
  test('Test for status value', () => {
    const StatusProps: StatusPropsType = {
      value: true
    };

    render(<Status {...StatusProps} />);
    const element = screen.getByTestId('status-test');

    if (element) expect(element).toHaveTextContent('active');
  });

  test('Test for snapshot rendered', () => {
    const StatusProps: StatusPropsType = {
      value: true
    };

    render(<Status {...StatusProps} />);
    const element = screen.getByTestId('status-test');

    expect(element).toMatchSnapshot();
  });
});
