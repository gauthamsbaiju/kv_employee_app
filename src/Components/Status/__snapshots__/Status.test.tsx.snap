// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for status Test for snapshot rendered 1`] = `
<div
  class="active"
  data-testid="status-test"
>
  active
</div>
`;
